lib/diagnostic.rb

```diff
@@ -7,36 +7,36 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    longest.map(&:length)
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    sum.reduce(0, :+)
```

Comment:
What does `sum.reduce` do? First of all, what is `sum`? It is a method that you have defined. So, in effect, you are calling your method from inside your method...

Secondly, notice that there is argument being passed into your function. Are you using that argument at all? 

Here's 1 solution:
```rb
  def sum(list)
    list.reduce(&:+)
  end
```